### PR TITLE
Improves macOS and iOS detection and version detection

### PR DESCRIPTION
**Pull request by liviuconcioiu**

### CODE REVIEW

**Comment by sanchezzzhak** [comment]
https://en.wikipedia.org/wiki/Darwin_(operating_system)#Release_history
we can detectiob ios version by Darwin

File: Tests/Parser/fixtures/oss.yml [hunk 1]
```diff
@@ -1108,3 +1098,185 @@
     short_name: DEE
     version: "15.11"
     platform: x64
+-
+  user_agent: Safari/11603.1.30.0.34 CFNetwork/760.6.3 Darwin/15.6.0 (x86_64)
+  os:
+    name: Mac
+    short_name: MAC
+    version: "10.11"
```
Comment (sanchezzzhak) [inline Tests/Parser/fixtures/oss.yml:1106]: 10.11.6

**Review by sanchezzzhak — DISMISSED** [review]

**Comment by liviuconcioiu** [comment]
Yes, I will do them later today.

**Comment by liviuconcioiu** [comment]
@sanchezzzhak can you please check those changes when you have time? If there are okay, I'll make changes for the other versions and for Mac too.

File: Tests/Parser/fixtures/oss.yml [hunk 2]
```diff
@@ -1115,3 +1105,262 @@
...
+    name: iOS
+    short_name: IOS
+    version: "12.4"
+    platform: ""
+-
+  user_agent: MobileSafari/604.1 CFNetwork/1125 Darwin/19.0.0
+  os:
+    name: iOS
+    short_name: IOS
+    version: "13"
```
Comment (sanchezzzhak) [inline Tests/Parser/fixtures/oss.yml:1316]: version: "13.0"

File: Tests/Parser/fixtures/oss.yml [hunk 3]
```diff
@@ -1115,3 +1105,262 @@
...
+    name: iOS
+    short_name: IOS
+    version: "13.6"
+    platform: ""
+-
+  user_agent: MobileSafari/604.1 CFNetwork/1197 Darwin/20.0.0
+  os:
+    name: iOS
+    short_name: IOS
+    version: "14"
```
Comment (sanchezzzhak) [inline Tests/Parser/fixtures/oss.yml:1358]: version: "14.0"

File: regexes/oss.yml [hunk 4]
```diff
@@ -303,23 +303,75 @@
 ##########
 # iOS
 ##########
-- regex: 'CFNetwork/889'
+- regex: 'CFNetwork/1206.*Darwin/20.1.0(?!.*x86_64)'
```
Comment (sanchezzzhak) [inline regexes/oss.yml:306]: cfnetwork version is no needed, change to
'CFNetwork/.+ Darwin/20.1.0(?!.*x86_64)'

File: regexes/oss.yml [hunk 5]
```diff
@@ -303,23 +303,75 @@
...
   name: 'iOS'
-  version: '11.1'
+  version: '14.2'
 
-- regex: 'CFNetwork/887.*(x86_64)'
-  name: 'Mac'
-  version: '10.13'
+- regex: 'CFNetwork/1197.*Darwin/20.0.0(?!.*x86_64)'
+  name: 'iOS'
+  version: '14'
```
Comment (sanchezzzhak) [inline regexes/oss.yml:312]: version: '14.0'

File: regexes/oss.yml [hunk 6]
```diff
@@ -303,23 +303,75 @@
...
+  name: 'iOS'
+  version: '12.0'
+
+- regex: 'CFNetwork/902.*Darwin/17.7.0(?!.*x86_64)'
+  name: 'iOS'
+  version: '11.4.1'
+
+- regex: 'CFNetwork/90[12].*Darwin/17.6.0(?!.*x86_64)'
+  name: 'iOS'
+  version: '11.4'
```
Comment (sanchezzzhak) [inline regexes/oss.yml:364]: replace  CFNetwork/* version if  definition goes by  darwin version
Example
```
CFNetwork/.+ Darwin/17.6.0(?!.*x86_64)
```

**Review by sanchezzzhak — CHANGES_REQUESTED** [review]

**Comment by sanchezzzhak** [comment]
add detect version `Safari/15610.2.11.51.10 CFNetwork/1209 Darwin/20.2.0`
macOS 11.1 
iOS 14.3

**Review by sanchezzzhak — DISMISSED** [review]

**Review by sanchezzzhak — DISMISSED** [review]

**Comment by liviuconcioiu** [comment]
I've found https://www.tonymacx86.com/ to be a more reliable source than Wikipedia. It has all versions.

`"macOS 10.12.6 Update" site:tonymacx86.com`
https://www.tonymacx86.com/threads/macos-10-12-6-update.224836/

**Review by sanchezzzhak — APPROVED** [review]

**Review by sgiehl — APPROVED** [review]

**Comment by adityapatadia** [comment]
Which version will have this capability?

**Comment by sanchezzzhak** [comment]
> Which version will have this capability?

available in the latest release 4.1

**Comment by adityapatadia** [comment]
When I add this user-agent, it does not detect the app name and version: `Clovia/3.3.1 CFNetwork/1206 Darwin/20.1.0`. We were expecting it to detect Clovia as client app with 3.3.1 version.

**Pull request merged**